monitor/index.ts

```diff
@@ -122,47 +128,82 @@ function notifyWhenNodesSleepingLongOrRecovered(
   bestBlockNumber: number,
   targetEmail: string,
   state: CheckSealFieldState,
-  sleepingNodeIndices: number[]
+  sleepingValidators: PlatformAddress[]
 ) {
-  const longTermSleepingIndices = [];
-  for (let idx = 0; idx < state.sleepStreak.length; idx++) {
-    if (sleepingNodeIndices.includes(idx)) {
-      state.sleepStreak[idx] += 1;
-    } else {
-      const sleepStreak = state.sleepStreak[idx];
-      const prevProblematic = sleepStreak >= state.sleepStreakAlertLevel;
-      if (prevProblematic) {
-        sendNotice(
-          new Notifications.NodeRecovered(bestBlockNumber, idx, sleepStreak),
-          targetEmail
-        );
+  const longTermSleepingAddresses: string[] = [];
+
+  const newSleepStreakState: { [index: string]: number } = {};
+
+  for (const sleepingValidator of sleepingValidators) {
+    newSleepStreakState[sleepingValidator.toString()] = 1;
+  }
+
+  for (const address in state.sleepStreak) {
+    if (state.sleepStreak.hasOwnProperty(address)) {
+      const prevSleepStreak = state.sleepStreak[address];
+      if (newSleepStreakState[address] === 0) {
```

Comment:
=== 0 check is not appropriate in this context.

Reply:
Thanks! I fixed it.